x-eithert/src/X/Control/Monad/Trans/Either.hs

```diff
@@ -59,6 +61,10 @@ eitherT f g m =
   either f g =<< runEitherT m
 {-# INLINE eitherT #-}
 
+tryEitherT :: (MonadCatch m, Exception e) => (e -> x) -> m a -> EitherT x m a
```

Comment:
Making it `Disaster -> x` would make `x-eithert` depend on `x-exception`. Is that ok?

Also for my use case, `Disaster` has no advantage over `Exception` because the monad action I'm using isn't one of ours and throws `Exception`, not `Disaster`.